macos/FluentUI/FilledTemplateImageView/FilledTemplateImageView.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+/// Custom view for displaying a template style image that can be filled with a specified color. This takes in an image with a corresponding
+/// overlay mask that is used for drawing the fill. The image and its corresponding fillMask need to line up appropriately so that when
+/// the image is drawn over it's mask, its outline will perfectly overlap the edges of the mask to achieve the desired fill effect.
+@objc(MSFFilledTemplateImageView)
+open class FilledTemplateImageView: NSImageView {
+
+	/// Creates a custom template style image view with a fill
+	/// - Parameters:
+	///   - image: the template style image or icon to be drawn
+	///   - fillMask: the  mask image used to draw the fill color
+	///   - contentTintColor: the color to use for the main image outline
+	///   - fillColor: the color to use for the image fill
+	@objc(initWithImage:fillMask:contentTintColor:FillColor:)
+	public init(
+		image: NSImage,
+		fillMask: NSImage,
+		contentTintColor: NSColor,
+		fillColor: NSColor
+	) {
+		self.fillMask = fillMask
+		self.fillColor = fillColor
+		super.init(frame: .zero)
+		self.image = image
+		self.contentTintColor = contentTintColor
+	}
+
+	@available(*, unavailable) required public init?(coder: NSCoder) {
+		preconditionFailure()
+	}
+	@available(*, unavailable) override init(frame: NSRect) {
+		preconditionFailure()
+	}
+
+	/// The  mask overlay image used to draw the fill.
+	@objc public var fillMask: NSImage {
+		didSet {
+			guard oldValue != fillMask else {
+				return
+			}
+			needsDisplay = true
+		}
+	}
+
+	/// The color used to draw the fill of the fillMask image,.
+	@objc public var fillColor: NSColor {
+		didSet {
+			guard oldValue != fillColor else {
+				return
+			}
+			needsDisplay = true
+		}
+	}
+
+	/// Public override used to draw the icon with the right border and fill
+	public override func draw(_ dirtyRect: NSRect) {
+
+		// Draw the image using the specified fill and border color
+		// Don't use a fillMask if it's set to clear
+		if fillColor != NSColor.clear {
```

Comment:
Any reason we use `NSColor.clear` instead of an optional value?

Reply:
Was on the fence on that one, but went with clear as I wanted to assume non null for everything.

Reply:
Note that we also don't want to draw with the clear color as it produces undesired effects with the transparency layer (shows up in black). So I kept that check there in addition to using an optional.